### ISSUE TITLE
Steam detached games failing to download.

### ISSUE THREAD

**Issue by orbea** [lutris/lutris#197]
I tried some of the steam detached installers, examples:

https://lutris.net/games/unreal-gold/
https://lutris.net/games/unreal-tournament/
https://lutris.net/games/quake-2/

Specifically I tried the two unreal installers, but it gets stuck when trying to download the actual game data saying, "Unable to get Steam data for game". 

```
INFO     2015-06-14 08:34:43,832 [installer]:Downloading file 1 of 3
DEBUG    2015-06-14 08:34:43,832 [installer]:Fetching [ut-bin]: http://lutris.net/files/games/unreal-tournament/unreal.tournament_436-multilanguage.goty.run
DEBUG    2015-06-14 08:34:43,833 [installer]:Downloading http://lutris.net/files/games/unreal-tournament/unreal.tournament_436-multilanguage.goty.run to /home/orbea/.cache/lutris/installer/unreal-tournament/unreal.tournament_436-multilanguage.goty.run
DEBUG    2015-06-14 08:34:45,351 [downloader]:Scheduling download
INFO     2015-06-14 08:35:04,770 [installer]:Downloading file 2 of 3
DEBUG    2015-06-14 08:35:04,771 [installer]:Fetching [utpg-patch]: http://lutris.net/files/games/unreal-tournament/UTPGPatch451.tar.bz2
DEBUG    2015-06-14 08:35:04,771 [installer]:Downloading http://lutris.net/files/games/unreal-tournament/UTPGPatch451.tar.bz2 to /home/orbea/.cache/lutris/installer/unreal-tournament/UTPGPatch451.tar.bz2
DEBUG    2015-06-14 08:35:05,118 [downloader]:Scheduling download
INFO     2015-06-14 08:35:13,781 [installer]:Downloading file 3 of 3
WARNING  2015-06-14 08:35:13,794 [steam]:Data path for SteamApp 13240 not found.
DEBUG    2015-06-14 08:35:13,795 [installer]:Installing steam game 13240
DEBUG    2015-06-14 08:35:14,096 [jobs]:Async call: install_game
DEBUG    2015-06-14 08:35:14,096 [steam]:Installing steam game 13240
ERROR    2015-06-14 08:35:14,097 [jobs]:Error while completing task <bound method steam.install_game of <lutris.runners.steam.steam object at 0x7efe8a077310>>: [Errno 2] No such file or directory: '/home/orbea/.local/share/Steam/SteamApps/appmanifest_13240.acf'
DEBUG    2015-06-14 08:35:16,101 [installer]:Steam game installed
WARNING  2015-06-14 08:35:16,102 [steam]:Data path for SteamApp 13240 not found.
ERROR    2015-06-14 08:35:16,102 [installer]:Unable to get Steam data for gameNone
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
```

Edit:  Decided to try Quake II also:

```
INFO     2015-06-14 09:21:45,103 [installer]:Downloading file 1 of 2
DEBUG    2015-06-14 09:21:45,103 [installer]:Fetching [yamagi-quake2-x64]: http://lutris.net/files/games/quake-2/yamagi-quake2-x64.tgz
DEBUG    2015-06-14 09:21:45,104 [installer]:Downloading http://lutris.net/files/games/quake-2/yamagi-quake2-x64.tgz to /home/orbea/.cache/lutris/installer/quake-2/yamagi-quake2-x64.tgz
DEBUG    2015-06-14 09:21:46,611 [downloader]:Scheduling download
INFO     2015-06-14 09:21:49,434 [installer]:Downloading file 2 of 2
DEBUG    2015-06-14 09:21:49,434 [installer]:Getting Wine Steam data for appid 2320
WARNING  2015-06-14 09:21:49,476 [winesteam]:Data path for SteamApp 2320 not found.
DEBUG    2015-06-14 09:21:49,476 [installer]:Installing steam game 2320
DEBUG    2015-06-14 09:21:49,777 [jobs]:Async call: install_game
DEBUG    2015-06-14 09:21:49,781 [winesteam]:Thread running: "/home/orbea/.local/share/lutris/runners/wine/1.7.29-i386/bin/wine" "/home/orbea/.local/share/lutris/runners/winesteam/prefix/drive_c/Program Files/Steam/Steam.exe" -no-dwrite steam://install/2320
preloader: Warning: failed to reserve range 00010000-00110000
wine client error:0: version mismatch 447/457.
Your wineserver binary was not upgraded correctly,
or you have an older one somewhere in your PATH.
Or maybe the wrong wineserver is still running?
DEBUG    2015-06-14 09:21:50,846 [installer]:Steam game installed
WARNING  2015-06-14 09:21:50,859 [winesteam]:Data path for SteamApp 2320 not found.
ERROR    2015-06-14 09:21:50,859 [installer]:Unable to get Steam data for gameNone
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

**Comment by orbea** [comment]
Well, apparently it was not downloading because I had another wine program running and closing it resolved downloading the game.  Unfortunately that was not enough to completely fix the issue...
I am getting an "OSError(8, 'Exec format error')" error now, here is the output.

~~~
lutris -d
DEBUG    2015-06-14 09:38:48,978 [lutris]:Welcome to Lutris
DEBUG    2015-06-14 09:38:49,055 [lutriswindow]:Getting game list
DEBUG    2015-06-14 09:38:49,333 [lutriswindow]:Switching view
DEBUG    2015-06-14 09:38:49,538 [lutriswindow]:Connecting signals
INFO     2015-06-14 09:38:49,636 [lutriswindow]:Connected as orbea
DEBUG    2015-06-14 09:38:49,638 [jobs]:Async call: sync_all
DEBUG    2015-06-14 09:38:49,638 [sync]:Syncing game library
DEBUG    2015-06-14 09:38:49,638 [jobs]:Async call: update_runtime
DEBUG    2015-06-14 09:38:49,638 [sync]:132 games in local library
DEBUG    2015-06-14 09:38:49,639 [api]:Fetching game library
DEBUG    2015-06-14 09:38:51,065 [sync]:132 games in remote library (inc. unpublished)
DEBUG    2015-06-14 09:38:51,095 [sync]:0 games updated
DEBUG    2015-06-14 09:38:51,095 [sync]:Syncing local steam games
DEBUG    2015-06-14 09:38:51,378 [jobs]:Async call: sync_icons
DEBUG    2015-06-14 09:38:51,380 [resources]:Fetching icons
Error creating proxy: Error calling StartServiceByName for org.gtk.Private.AfcVolumeMonitor: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.gtk.Private.AfcVolumeMonitor exited with status 127 (g-dbus-error-quark, 25)
INFO     2015-06-14 09:38:59,061 [installer]:Downloading file 1 of 2
DEBUG    2015-06-14 09:38:59,061 [installer]:Fetching [liflg-installer]: http://liflg.org/torrents/unreal.tournament_436-multilanguage.goty.run.torrent
DEBUG    2015-06-14 09:38:59,063 [installer]:Downloading http://liflg.org/torrents/unreal.tournament_436-multilanguage.goty.run.torrent to /home/orbea/.cache/lutris/installer/unreal-gold/unreal.tournament_436-multilanguage.goty.run.torrent
DEBUG    2015-06-14 09:38:59,910 [downloader]:Scheduling download
INFO     2015-06-14 09:39:00,312 [installer]:Downloading file 2 of 2
DEBUG    2015-06-14 09:39:00,312 [installer]:Getting Wine Steam data for appid 13250
DEBUG    2015-06-14 09:39:00,354 [steam]:Game 13250 should be in Unreal Gold
DEBUG    2015-06-14 09:39:00,367 [steam]:Game 13250 should be in Unreal Gold
DEBUG    2015-06-14 09:39:00,367 [installer]:got data path: /media/data/home/games/Lutris/steamapps/common/Unreal Gold
DEBUG    2015-06-14 09:39:00,368 [jobs]:Async call: execute
DEBUG    2015-06-14 09:39:00,378 [installer]:Executing /home/orbea/.cache/lutris/installer/unreal-gold/unreal.tournament_436-multilanguage.goty.run.torrent []
ERROR    2015-06-14 09:39:00,385 [jobs]:Error while completing task <bound method ScriptInterpreter.execute of <lutris.installer.ScriptInterpreter object at 0x7fa471776e90>>: [Errno 8] Exec format error

**Comment by orbea** [comment]
I tried Quake II again, that actually installs without issue, but won't start.

~~~
$ lutris -d
DEBUG    2015-06-14 09:45:53,307 [lutris]:Welcome to Lutris
DEBUG    2015-06-14 09:45:53,418 [lutriswindow]:Getting game list
DEBUG    2015-06-14 09:45:53,699 [lutriswindow]:Switching view
DEBUG    2015-06-14 09:45:53,900 [lutriswindow]:Connecting signals
INFO     2015-06-14 09:45:53,989 [lutriswindow]:Connected as orbea
DEBUG    2015-06-14 09:45:53,991 [jobs]:Async call: sync_all
DEBUG    2015-06-14 09:45:53,991 [sync]:Syncing game library
DEBUG    2015-06-14 09:45:53,991 [jobs]:Async call: update_runtime
DEBUG    2015-06-14 09:45:53,992 [sync]:132 games in local library
DEBUG    2015-06-14 09:45:53,992 [api]:Fetching game library
DEBUG    2015-06-14 09:45:55,408 [sync]:132 games in remote library (inc. unpublished)
DEBUG    2015-06-14 09:45:55,438 [sync]:0 games updated
DEBUG    2015-06-14 09:45:55,439 [sync]:Syncing local steam games
DEBUG    2015-06-14 09:45:55,728 [jobs]:Async call: sync_icons
DEBUG    2015-06-14 09:45:55,730 [resources]:Fetching icons
DEBUG    2015-06-14 09:46:05,361 [game]:Launching Quake 2: {'command': ['"/home/orbea/games/Lutris/quake-2/quake2"']}
DEBUG    2015-06-14 09:46:05,361 [display]:Switching resolution to 1680x1050
DEBUG    2015-06-14 09:46:05,436 [thread]:Running thread from /home/orbea/games/Lutris/quake-2
DEBUG    2015-06-14 09:46:05,437 [thread]:Thread running: LD_LIBRARY_PATH="/home/orbea/.local/share/lutris/runtime/lib64:/home/orbea/.local/share/lutris/runtime/lib32:$LD_LIBRARY_PATH" "/home/orbea/games/Lutris/quake-2/quake2"
Inconsistency detected by ld.so: dl-open.c: 680: _dl_open: Assertion `_dl_debug_initialize (0, args.nsid)->r_state == RT_CONSISTENT' failed!
DEBUG    2015-06-14 09:46:07,439 [thread]:Game quit
DEBUG    2015-06-14 09:46:10,438 [game]:game has quit at Sun, 14 Jun 2015 09:46:10
DEBUG    2015-06-14 09:46:10,439 [display]:Switching to 1680x1050+0+0 on DVI-I-2
DEBUG    2015-06-14 09:46:10,544 [display]:xrandr --output DVI-I-2 --mode 1680x1050 --pos 0x0
DEBUG    2015-06-14 09:46:10,548 [thread]:SIGKILL 3153
ERROR    2015-06-14 09:46:10,548 [thread]:Could not kill PID 3153
~~~

While Unreal Tournament remains stuck in the same place as before unable to download the game.

~~~
$ lutris -d
DEBUG    2015-06-14 09:47:09,089 [lutris]:Welcome to Lutris
DEBUG    2015-06-14 09:47:09,168 [lutriswindow]:Getting game list
DEBUG    2015-06-14 09:47:09,454 [lutriswindow]:Switching view
DEBUG    2015-06-14 09:47:09,658 [lutriswindow]:Connecting signals
INFO     2015-06-14 09:47:09,752 [lutriswindow]:Connected as orbea
DEBUG    2015-06-14 09:47:09,753 [jobs]:Async call: sync_all
DEBUG    2015-06-14 09:47:09,754 [sync]:Syncing game library
DEBUG    2015-06-14 09:47:09,754 [jobs]:Async call: update_runtime
DEBUG    2015-06-14 09:47:09,754 [sync]:132 games in local library
DEBUG    2015-06-14 09:47:09,754 [api]:Fetching game library
DEBUG    2015-06-14 09:47:11,146 [sync]:132 games in remote library (inc. unpublished)
DEBUG    2015-06-14 09:47:11,177 [sync]:0 games updated
DEBUG    2015-06-14 09:47:11,177 [sync]:Syncing local steam games
DEBUG    2015-06-14 09:47:11,477 [jobs]:Async call: sync_icons
DEBUG    2015-06-14 09:47:11,479 [resources]:Fetching icons
Error creating proxy: Error calling StartServiceByName for org.gtk.Private.AfcVolumeMonitor: GDBus.Error:org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.gtk.Private.AfcVolumeMonitor exited with status 127 (g-dbus-error-quark, 25)
INFO     2015-06-14 09:47:20,912 [installer]:Downloading file 1 of 3
DEBUG    2015-06-14 09:47:20,912 [installer]:Fetching [ut-bin]: http://lutris.net/files/games/unreal-tournament/unreal.tournament_436-multilanguage.goty.run
DEBUG    2015-06-14 09:47:20,913 [installer]:Downloading http://lutris.net/files/games/unreal-tournament/unreal.tournament_436-multilanguage.goty.run to /home/orbea/.cache/lutris/installer/unreal-tournament/unreal.tournament_436-multilanguage.goty.run
DEBUG    2015-06-14 09:47:22,176 [downloader]:Scheduling download
INFO     2015-06-14 09:47:27,827 [installer]:Downloading file 2 of 3
DEBUG    2015-06-14 09:47:27,827 [installer]:Fetching [utpg-patch]: http://lutris.net/files/games/unreal-tournament/UTPGPatch451.tar.bz2
DEBUG    2015-06-14 09:47:27,827 [installer]:Downloading http://lutris.net/files/games/unreal-tournament/UTPGPatch451.tar.bz2 to /home/orbea/.cache/lutris/installer/unreal-tournament/UTPGPatch451.tar.bz2
DEBUG    2015-06-14 09:47:28,181 [downloader]:Scheduling download
INFO     2015-06-14 09:47:32,218 [installer]:Downloading file 3 of 3
WARNING  2015-06-14 09:47:32,219 [steam]:Data path for SteamApp 13240 not found.
DEBUG    2015-06-14 09:47:32,219 [installer]:Installing steam game 13240
DEBUG    2015-06-14 09:47:32,521 [jobs]:Async call: install_game
DEBUG    2015-06-14 09:47:32,521 [steam]:Installing steam game 13240
ERROR    2015-06-14 09:47:32,521 [jobs]:Error while completing task <bound method steam.install_game of <lutris.runners.steam.steam object at 0x7fd271556ed0>>: [Errno 2] No such file or directory: '/home/orbea/.local/share/Steam/SteamApps/appmanifest_13240.acf'
DEBUG    2015-06-14 09:47:49,039 [installer]:Steam game installed
WARNING  2015-06-14 09:47:49,040 [steam]:Data path for SteamApp 13240 not found.
ERROR    2015-06-14 09:47:49,040 [installer]:Unable to get Steam data for gameNone
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

**Comment by strycore** [comment]
Let me check on that, something is very wrong here…
I tried installing Unreal Tournament and it worked because I already had the steam data available, the Unreal Gold has been altered and the current doesn't make any kind of sense (tries to download a torrent file and downloads UT data files instead of Unreal).

The whole steam data download process seems to be broken, it starts multiple instances of Steam until the process is killed, probably due to the changes we made to process management I guess.

Once I sort that out, I'll make sure that all installers use native steam (as there is no reason to use winesteam anymore).

**Comment by strycore** [comment]
Reverted a strange commit (c161b4da3ed4ef88310b76d7e33a1b9f9dd0ca50) fixing the issue of the infinite loop. I successfully installed Quake 2.

I have yet to rewrite the unreal gold installer (or maybe write it? I may have never finished it to begin with…) and also rename the installers because I don't want "Steam detached" to become a thing :p (I find it very confusing and I'd rather refer to those installers as "Native + Steam data")
